dataflux_pytorch/benchmark/checkpointing/multinode/README.md

```diff
@@ -4,56 +4,115 @@ This benchmarking script will allow you to run and benchmark the performance of
+  * `NUM_SAVE_CALLS`: The number of times `trainer.save_checkpoint` is called. Defaults to `3`.
   
-  * `NUM_LOAD_CALLS`: The number of times `trainer.strategy.load_checkpoint` is called. Defaults to 3. 
+  * `NUM_LOAD_CALLS`: The number of times `trainer.strategy.load_checkpoint` is called. Defaults to `3`. 
 
-  * `NUM_NODES`: The number of nodes (machines). Defaults to 1.
+  * `NUM_NODES`: The number of nodes you wish to deploy the workload on. Defaults to `1`.
 
-  * `NUM_DEVICES`: The number of devices per node, or `auto`. Defaults to `auto`. 
+  * `NUM_DEVICES`: The number of devices per node. Defaults to `auto`. 
 
+### Local Execution
 
-### Installing Requirements:
- 
-* Install the requirements using following command `pip install -r dataflux_pytorch/benchmark/requirements.txt`; `pip install .`
+#### Dataflux
+```shell
+python3 -m dataflux_pytorch.benchmark.checkpointing.multinode.train --strategy=dataflux_fsdp
+```
 
+#### FSSpec
+```shell
+python3 -m dataflux_pytorch.benchmark.checkpointing.multinode.train --strategy=fsspec_fsdp
+```
 
-### Checkpointing Strategy: 
+#### Boot Disk
+```shell
+# Benchmark checkopint saves first
+python3 -m dataflux_pytorch.benchmark.checkpointing.multinode.train --strategy=fsdp --save_only
 
-* Custom Strategy needs to be implemented and passed to the trainer. The strategy needs to extend [Strategy class](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.strategies.Strategy.html) or one of the class which extend Strategy class. When using Strategy which extends [FSDPStrategy](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.strategies.FSDPStrategy.html), GPU is required.
+# Benchmark loads next
+python3 -m dataflux_pytorch.benchmark.checkpointing.multinode.train --strategy=fsdp --load_only
+```
 
-* According to Pytorch lightning documentation for FSDPStrategy the model is divided accross all the available GPU's accross all the nodes, so num_nodes can be >=1. The only caveat is to make sure that num_nodes * devices should be equal total available number of GPU. For example if you have 1 VM with 4 GPU you can or 4 VM's with 1 GPU each. num_nodes and devices can be either 1 and 4 respectively or 4 and 1 respectively, in either case FSDP will write 1 shard per GPU.
+#### Distributes Filesystem
+```shell
+python3 -m dataflux_pytorch.benchmark.checkpointing.multinode.train --strategy=fsdp --distributed_filesystem
+```
 
-### Running
+### Multi-node Execution
 
-To run the script use the following command. 
+_Note: the following instructions assume that you have Jobset and Kueue enabled on your GKE cluster. For easy compatability we recommend creating a cluster with [XPK](https://github.com/google/xpk) which will configure these features automatically._
 
-```shell
-python3 -m dataflux_pytorch.benchmark.checkpointing.multinode.train
-```
+1. Connect to your GKE cluster from your workstation
```

Comment:
These all have a 1 in front of them. Also do we want a step for making modifications to the yaml file? We would need to sync some params for deployment right?

Reply:
Markdown magic! MD automatically numbers the list items so we don't have to update them when we insert/delete a list item.